File: src/features/email-templates/dto/update-email-template.dto.ts
import { IsOptional, IsString } from 'class-validator';

export class UpdateEmailTemplateDto {
  @IsOptional()
  @IsString()
  locale?: string;

  @IsString()
  subject: string;

  @IsString()
  html: string;

  @IsString()
  text: string;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/emailtemplates/UpdateEmailTemplateDto.java
package com.mytegroup.api.dto.emailtemplates;

import jakarta.validation.constraints.NotBlank;

public record UpdateEmailTemplateDto(
    String locale,
    @NotBlank(message = "Subject is required")
    String subject,
    @NotBlank(message = "HTML content is required")
    String html,
    @NotBlank(message = "Text content is required")
    String text
) {
    public UpdateEmailTemplateDto {
        if (locale != null) {
            locale = locale.trim();
        }
        if (subject != null) {
            subject = subject.trim();
        }
    }
}



